pkg/cloudprovider/aws/factory.go

```diff
@@ -92,6 +92,12 @@ func (f *Factory) CapacityFor(provisioner *v1alpha1.Provisioner) cloudprovider.C
 	}
 }
 
+func (f *Factory) TerminateFor() cloudprovider.Termination {
```

Comment:
Why separate this to a different interface? In my mind, it's all the capacity interface on the cloud provider.

Reply:
We don't need the provisioner to terminate nodes, so I separated it out into its own Factory implementation.

Reply:
I see the challenge here, but we need a cleaner solution to this.